### ISSUE TITLE
[9.0] Selecting accounts with movements in Trial balance doesn't work as expected

### ISSUE THREAD

**Issue by MouTio** [odoo/odoo#17185]
Impacted versions:
Odoo 9

Steps to reproduce:
In Accounting -> Reports -> Trial Balance, I select a period of one year and then I click on show accounts "with movements".

Current behavior:
In the report only appears the accounts with total balance not equal to zero.

Expected behavior:
When I check the show accounts "with movements" option, I expect that the trial balance report have all the accounts with movements in the period, that is all the accounts with debit or credit not equal to zero.

Instead of this (in account/report/account_balance.py):

```
if display_account == 'all':
    account_res.append(res)
if display_account in ['movement', 'not_zero'] and not currency.is_zero(res['balance']):
    account_res.append(res)
```

The working code will be something like this:

```
if display_account == 'all':
    account_res.append(res)
if display_account == 'not_zero' and not currency.is_zero(res['balance']):
    account_res.append(res)
if display_account == 'movement' and (not currency.is_zero(res['debit']) or not currency.is_zero(res['credit'])):
    account_res.append(res)
```

**Comment by Vartkat** [comment]
Hi,

Same in v10ce, with movements gives the same result as if I had chosen with balance not at zero. When asking for "with movements" none of the accounts has a balance at 0,00.

V.

**Comment by Yenthe666** [comment]
@qdp-odoo what do you think about the change? If you approve I'll make a PR for you. Just let me know.

**Comment by Vartkat** [comment]
Sorry I will look dumb but I don't know where's the code, how to pull it and merge it with my test config to test it. Can you guide me ?

**Comment by qdp-odoo** [comment]
I confirm that this is a bug.

@Yenthe666 you can proceed to the PR if you want :-) Thanks a lot

**Comment by qdp-odoo** [comment]
fix landed on revision https://github.com/odoo/odoo/commit/d6751e5b6ae9cfdcc9ce46ec89106c5d27281ca2

Nice work @Yenthe666 :-)

**Comment by Yenthe666** [comment]
Thank you! The idea / credit goes to @MouTio though. 😉 
Thanks for handeling the report so fast!

**Comment by Vartkat** [comment]
@Yenthe666. I’ve copy/pasted your modification in a 10.0. It works so you can do it on 10 also.

Thanks you very much.

V.

**Comment by Yenthe666** [comment]
@Vartkat it has been merged and fixed in 9.0 and will be automatically forwarded to Odoo 10. This usually takes about a week.
